bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerV2Handler.java

```diff
@@ -322,12 +325,26 @@ public void initialize() {
     }
 
     private String getReadUrl() {
-        return GoEChargerBindingConstants.API_URL_V2.replace("%IP%", config.ip.toString()) + filter;
+        if (config.ip != null) {
+            return GoEChargerBindingConstants.API_URL_V2.replace("%IP%", config.ip.toString()) + "?" + filter;
+        } else if (config.serial != null && config.token != null) {
+            return GoEChargerBindingConstants.API_URL_CLOUD_V2.replace("%SERIAL%", config.serial.toString())
+                    .replace("%TOKEN%", config.token.toString()) + "&" + filter;
+        } else {
+            throw new IllegalArgumentException("either ip or token+serial must be configured");
+        }
     }
 
     private String getWriteUrl(String key, String value) {
```

Comment:
As this change adds a `IllegalArgumentException` the method should be annotated with this exception so callers are forced to handle this case.